### PR TITLE
Phase 1 Deployment Ownership & Deposits

### PR DESCRIPTION
**Pull request by chknlittle**
- Adds deposit to each vault
- Adds transfer of DEFAULT_ADMIN_ROLE from deployment address to multisig

### CODE REVIEW

File: script/DeployLeveragedEthMainnet.s.sol [hunk 1]
```diff
@@ -45,23 +53,69 @@ contract DeployScript is CREATE3Script {
...
+        deposit(scUsdc, usdc, usdc.balanceOf(address(deployerAddress))); // 0.01 ether worth of USDC
+
+        // transfer DEFAULT_ADMIN_ROLE to multisig
+        scUsdc.grantRole(DEFAULT_ADMIN_ROLE, C.MULTISIG);
+        scUsdc.revokeRole(DEFAULT_ADMIN_ROLE, deployerAddress);
+
         vm.stopBroadcast();
     }
+
+    function deposit(sc4626 vault, ERC20 underlying, uint256 amount) internal {
```
Comment (brkomir) [inline script/DeployLeveragedEthMainnet.s.sol:90]: you can call `vault.asset()` to get the underlying, no need to pass it as parameter

File: script/DeployLeveragedEthMainnet.s.sol [hunk 2]
```diff
@@ -45,23 +53,69 @@ contract DeployScript is CREATE3Script {
...
+
+        weth.approve(address(uniswapRouter), amount);
+
+        address[] memory tokenPath = new address[](2);
+        tokenPath[0] = address(weth);
+        tokenPath[1] = address(usdc);
+
+        uint256[] memory amountsMin = new uint256[](2);
+        amountsMin[0] = amount;
+        amountsMin[1] = 0;
```
Comment (brkomir) [inline script/DeployLeveragedEthMainnet.s.sol:106]: this seems unnecessary, right?

File: src/lib/Constants.sol [hunk 3]
```diff
@@ -51,4 +51,7 @@ library Constants {
 
     // address of the 0x swap router contract
     address public constant ZEROX_ROUTER = 0xDef1C0ded9bec7F1a1670819833240f027b25EfF;
+
+    // address of the multisig
+    address public constant MULTISIG = 0x035F210e5d14054E8AE5A6CFA76d643aA200D56E;
```
Comment (brkomir) [inline src/lib/Constants.sol:56]: maybe move this to the top of the library and add to the comment that this address supposed to have the admin role

File: script/DeployLeveragedEthMainnet.s.sol (hunk 2 above)
Comment (brkomir) [reply script/DeployLeveragedEthMainnet.s.sol:106]: I mean you are creating two arrays just to pass the values you assign to the struct initialized below, just use the values directly

Comment (0xrin1) [reply script/DeployLeveragedEthMainnet.s.sol]: so true :facepalm: removed

**Review by brkomir — APPROVED** [review]

**Pull request merged**